File: src/main/java/com/amazon/kinesis/kafka/config/ClusterMapping.java
package com.amazon.kinesis.kafka.config;

import java.util.Collections;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

public class ClusterMapping {

    private String clusterName;
    private List<DestinationStreamMapping> streams;

    public String getClusterName() {
        return clusterName;
    }

    public void setClusterName(String clusterName) {
        this.clusterName = clusterName;
    }

    public List<DestinationStreamMapping> getStreams() {
        return streams;
    }

    public void setStreams(List<DestinationStreamMapping> streams) {
        this.streams = streams;
    }

    public Map<String, List<String>> getStreamsAsMap() {
        if (streams == null) {
            return Collections.emptyMap();
        }

        return streams.stream()
                .collect(Collectors.toMap(DestinationStreamMapping::getName, DestinationStreamMapping::getDestinations));
    }

}

File: src/main/kotlin/com/amazon/kinesis/kafka/config/ClusterMapping.kt
package com.amazon.kinesis.kafka.config

class ClusterMapping {

    val clusterName: String = ""
    val streams: List<DestinationStreamMapping> = emptyList()

    val streamsAsMap: Map<String, List<String>?>
        get() = streams.associateBy( {it.name}, {it.destinations} )

    fun gatherStreamFilters(): Map<String, List<StreamFilterMapping>?> {
        return streams.associateBy( { it.name }, { it.filters })
    }
}
